File: Hospital Management System/backend/src/main/java/com/hospital/management/service/BillingService.java
package com.hospital.management.service;

import com.hospital.management.model.Billing;

import java.util.List;

public interface BillingService {
    
    Billing createBilling(Billing billing);
    
    List<Billing> getAllBillings();
    
    Billing getBillingById(Long id);
    
    Billing updateBilling(Long id, Billing billing);
    
    void deleteBilling(Long id);
    
    Double calculateTotalAmount(Double consultationFee, Double treatmentFee, Double medicineFee);
}

File: Hospital Management System/frontend/src/services/billingService.js
import api from './api';

const billingService = {
  getAllBillings: async () => {
    try {
      const response = await api.get('/billings');
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  addBilling: async (billing) => {
    try {
      const response = await api.post('/billings', billing);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  getBillingById: async (id) => {
    try {
      const response = await api.get(`/billings/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  updateBilling: async (id, billing) => {
    try {
      const response = await api.put(`/billings/${id}`, billing);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  deleteBilling: async (id) => {
    try {
      const response = await api.delete(`/billings/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },
};

export default billingService;

